ShoeBox/Classes/Helpers/LocationManager/LocationManager.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  LocationManager.swift
+//  ShoeBox
+//
+//  Created by Madalina Miron on 28/11/16.
+//  Copyright © 2016 ShoeBox. All rights reserved.
+//
+
+import Foundation
+import CoreLocation
+
+protocol LocationManagerDelegate {
+    func locationReceived(currentLocation: CLLocation)
+    func locationDidFailWithError(error: NSError)
+}
+
+class LocationManager: NSObject, CLLocationManagerDelegate {
+    static let sharedInstance = LocationManager()
+    
+    var locationManager: CLLocationManager?
+    var currentLocation: CLLocation?
+    var delegate: LocationManagerDelegate?
+    
+    override init() {
+        super.init()
+        
+        self.locationManager = CLLocationManager()
+        guard let locationManager = self.locationManager else {
```

Comment:
no need to check if locationManager exists. Just use  self.locationManager?.delegate = self and 
 self.locationManager?. startUpdatingLocation()